lib/core/include/irods/irods_load_plugin.hpp

```diff
@@ -94,6 +101,274 @@ namespace irods
 
     } // resolve_plugin_path
 
+    /**
+     * \brief attempt to parse DT_FLAGS_1 to identify flags to pass to dlopen()
+     *
+     * \since 4.3.1
+     **/
+    // NOLINTNEXTLINE(readability-function-cognitive-complexity) this is technically a parser
+    inline int get_plugin_rtld_flags(const std::string& _plugin_path) noexcept
+    {
+        using logger = irods::experimental::log::server;
+
+        int rtld_lazy_or_now = RTLD_LAZY;
+        int rtld_local_or_global = RTLD_LOCAL;
+        int rtld_nodelete = 0;
+
+        try {
+            std::ifstream lib(_plugin_path, std::ios::binary);
+            if (!lib) {
+                // clang-format off
+                logger::error({
+                    {"plugin_path", _plugin_path},
+                    {"message", "Could not parse DT_FLAGS_1: failed to open plugin file"}
```

Comment:
Our regular/default log messages use the key "log_message". Should we use the same key for consistency?

I realize that by now this may not be consistent everywhere, but figured I'd ask.

Reply:
Good catch. I used `log_message` in PREP so I'll use that here too.